src/resolver.rs

```diff
@@ -1177,7 +1177,8 @@ impl<'a> AuditGraph<'a> {
                         audit_index,
                         publisher_index,
                     };
-                    let is_fresh_import = entry.is_fresh_import || publisher.is_fresh_import;
+                    let is_fresh_import = import_index.is_some()
+                        && (entry.is_fresh_import || publisher.is_fresh_import);
```

Comment:
(I meant to include a comment like this when I first submitted the review, but apparently it didn't save - apologies)

We shouldn't consider a local wildcard import which has a fresh import to be a non-fresh import. Something like that requires changes to `imports.lock` in order to use, so should be considered fresh.

Perhaps we could change `DeltaEdge::is_fresh_import` to instead be an enum like:

```rust
#[derive(Debug, Copy, Clone, Eq, PartialEq)]
enum DeltaEdgeFreshness {
  // All information requried for this delta edge is already stored within the store's supply-chain.
  // Using this edge will require no changes to imports.lock.
  Stale,
  // This edge originates from an already-imported or local wildcard audit,
  // however will require importing fresh publisher information into imports.lock to use.
  FreshPublisher,
  // This edge is fully fresh, and will require adding the audit entry to imports.lock to use.
  Fresh,
}

impl DeltaEdgeFreshness {
  fn is_fresh(&self) -> bool { self != DeltaEdgeFreshness::Stale }
}
```

For most of the logic, checks of `is_fresh_import` can be replaced with the equivalent of `is_fresh()` (or just inlining that check), but the caveat level match pattern would need to be changed, like:

```rust
                _ => match edge.is_fresh_import {
                   DeltaEdgeFreshness::Stale => CaveatLevel::None,
                   DeltaEdgeFreshness::FreshPublisher => CaveatLevel::FreshPublisher,
                   DeltaEdgeFreshness::Fresh => CaveatLevel::FreshImport,
               }
```

Reply:
This is the change I was most hesitant about, so I'm glad you had some opinions on it.